xbmc/epg/EpgContainer.cpp

```diff
@@ -340,25 +341,31 @@ CEpg *CEpgContainer::GetById(int iEpgId) const
     return NULL;
 
   CSingleLock lock(m_critSection);
-  std::map<unsigned int, CEpg *>::const_iterator it = m_epgs.find((unsigned int) iEpgId);
-  return it != m_epgs.end() ? it->second : NULL;
+  const auto &epgEntry = m_epgs.find((unsigned int) iEpgId);
+  return epgEntry != m_epgs.end() ? epgEntry->second : NULL;
 }
 
 CEpgInfoTagPtr CEpgContainer::GetTagById(int iBroadcastId) const
 {
   CEpgInfoTagPtr retval;
   CSingleLock lock(m_critSection);
-  for (std::map<unsigned int, CEpg *>::const_iterator it = m_epgs.begin(); !retval && it != m_epgs.end(); ++it)
-    retval = it->second->GetTag(iBroadcastId);
+  for (const auto &epg : m_epgs)
+  {
+    retval = epg.second->GetTag(iBroadcastId);
+    if (retval)
```

Comment:
same comment as above really, the new notation splits up the for statement from the break condition, which is meh. just change it to use `EPGMAP_CITR` if you want to improve readability